src/main/java/gov/nasa/pds/tools/validate/rule/pds4/BundleReferentialIntegrityRule.java

```diff
@@ -88,12 +99,263 @@ public boolean isApplicable(String location) {
+
+                    // Only throw a WARNING if the product does belong to this bundle.
+                    if (productBelongToBundleFlag == true) {
+                        LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:PRODUCT_IS_IN_BUNDLE:filename,logicalIdentifierPerFilename {},{}",filename,logicalIdentifierPerFilename);
+                        this.performReporting(singleLidOrLidvidReference, false, indexToFilenames);
+                    } else {
+                        LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:PRODUCT_NOT_IN_BUNDLE:filename,logicalIdentifierPerFilename {},{}",filename,logicalIdentifierPerFilename);
+                    }
+                } else {
+                    LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:LID_REFERENCE:singleLidOrLidvidReference {} is in logicalIdentifiersCumulative",singleLidOrLidvidReference);
+                }
+            } else {
+                if (!this.logicalIdentifiersCumulative.contains(singleLidOrLidvidReference)) {
+
+                    // We also need to check if the product is actually a product in the bundle.
+                    // We should not throw a WARNING if the product does not belong to the bundle.
+                    String filename = this.lidOrLidVidReferencesCumulativeFileNames.get(indexToFilenames).toString();
+                    String logicalIdentifierPerFilename = this.filenameToLogicalIdentifierMap.get(filename);
+                    LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:filename,logicalIdentifierPerFilename {},{}",filename,logicalIdentifierPerFilename);
+
+                    boolean productBelongToBundleFlag = this.isIdentiferMatchingBundleBaseID(logicalIdentifierPerFilename);
+
+                    // Only throw a WARNING if the product does belong to this bundle.
+                    if (productBelongToBundleFlag == true) {
+                        LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:PRODUCT_IS_IN_BUNDLE:filename,logicalIdentifierPerFilename {},{}",filename,logicalIdentifierPerFilename);
+                        this.performReporting(singleLidOrLidvidReference, true, indexToFilenames);
+                    } else {
+                        LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:PRODUCT_NOT_IN_BUNDLE:filename,logicalIdentifierPerFilename {},{}",filename,logicalIdentifierPerFilename);
+                    }
+                } else {
+                    LOG.debug("reportLidOrLidvidReferenceToNonExistLogicalReferences:LIDVID_REFERENCE:singleLidOrLidvidReference {} is in logicalIdentifiersCumulative",singleLidOrLidvidReference);
+                }
+            }
+            indexToFilenames += 1;
+        } // end for loop
+      } catch (Exception e) {
+          e.printStackTrace();
+      }
+  }
+
+  private boolean hasReferenceIDAndFilenameComboAdded(String singleLidorLidVidReference, URL filename) {
```

Comment:
@qchaupds can we use some other data structure here instead of combining this info with a `slash`? this seems like we are trying to force this into a string, but you never know if we will need to utilize these data structures for something else in the future.

additionally, is there any way we can use the `lidvid` of the product in order to do these mappings instead of the filename? the filename will most likely work, but, the LIDVID is really the identifier for the product and avoids any possible weird collisions.

Reply:
The slash has been removed to avoid confusion.  We are trying to compare a combination of two strings so a simple concatenation was sufficient.

The mapping is now from a lid_reference or lidvid_refence to a logical_identifier (instead of from a filename).